streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -43,6 +43,8 @@
 
     private static final Logger log = LoggerFactory.getLogger(StreamTask.class);
 
+    private static final ConsumerRecord<Object, Object> DUMMY_RECORD = new ConsumerRecord<>("__null_topic__", -1, -1L, null, null);
```

Comment:
This doesn't seem to conform to the comments in `ProcessorContext`. Specifically, the topic is not null and the timestamp will be `NO_TIMESTAMP`. Nothing is wrong with that for `DUMMY_RECORD` itself, but it looks like `ProcessorContext` is implemented by `ProcessorContextImpl` and `StandbyContextImpl`. I only looked into `ProcessorContextImpl` so far, but the methods mentioned (along with `partition` and `offset`, which seem to behave properly) just forward the request to `currRecord` (which is `task.record()` in those classes). Since `currRecord = DUMMY_RECORD` during the `punctuate` call, they will return the wrong values.

I think we probably need unit tests for these methods during a call to punctuate to properly test this behavior.